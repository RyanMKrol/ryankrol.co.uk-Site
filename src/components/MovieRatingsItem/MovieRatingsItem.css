.MovieRatingsItem {
  padding: 15px;
  max-width: 300px;
  text-align: left;
  background-color: #eee;
  margin: 10px;
  padding-bottom: 0;
  font-family: 'Archivo Narrow', sans-serif;
  font-size: 14px;
  border-radius: 5px;
  display: flex;
  flex-direction: column;
}
