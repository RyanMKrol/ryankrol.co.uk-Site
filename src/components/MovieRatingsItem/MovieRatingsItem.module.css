.outerContainer {
  padding: 4px;
}

.container {
  display: flex;
  justify-content: center;
  align-items: center;

  width: 500px;

  background-color: rgba(255, 255, 255, 0.9);
  flex-direction: column;
}

.titleContent {
  margin-bottom: 4px;
}

.titleUnderline {
  border-top-style: solid;
  border-top-color: black;
  border-top-width: 1px;
  width: 100%;
  height: auto;
}

.content {
  display: flex;
  align-items: center;
}

.thumbnailContainer {
  justify-content: center;
  align-items: center;
  max-height: 160px;
}

.thumbnail {
  max-height: 160px;
}
