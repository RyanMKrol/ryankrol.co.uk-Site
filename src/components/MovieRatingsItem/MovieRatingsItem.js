import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './MovieRatingsItem.css';

/**
 * Class representing a UI component for an movie
 */
class MovieRatingsItem extends Component {
  /**
   * @param props
   */
  constructor(props) {
    super();

    this.title = props.title;
    this.blind = props.blind;
    this.characters = props.characters;
    this.craftsmanship = props.craftsmanship;
    this.gist = props.gist;
    this.sound = props.sound;
    this.story = props.story;

    if (
      !(
        typeof this.title !== 'undefined'
        && typeof this.blind !== 'undefined'
        && typeof this.characters !== 'undefined'
        && typeof this.craftsmanship !== 'undefined'
        && typeof this.gist !== 'undefined'
        && typeof this.sound !== 'undefined'
        && typeof this.story !== 'undefined'
      )
    ) {
      throw new Error('Did not pass all required args for rendering MovieRatingsItem');
    }
  }

  /**
   * Render method
   *
   * @returns {string} The JSX representing the component
   */
  render() {
    return (
      <div className="MovieRatingsItem">
        <p className="movie-ratings-item-content">
          <span className="movie-ratings-item-key">Title - </span>
          <span className="movie-ratings-item-value">{this.title}</span>
          <br />
          <span className="movie-ratings-item-key">Blind - </span>
          <span className="movie-ratings-item-value">{this.blind}</span>
          <br />
          <span className="movie-ratings-item-key">Characters - </span>
          <span className="movie-ratings-item-value">{this.characters}</span>
          <br />
          <span className="movie-ratings-item-key">Craftsmanship - </span>
          <span className="movie-ratings-item-value">{this.craftsmanship}</span>
          <br />
          <span className="movie-ratings-item-key">Sound - </span>
          <span className="movie-ratings-item-value">{this.sound}</span>
          <br />
          <span className="movie-ratings-item-key">Story - </span>
          <span className="movie-ratings-item-value">{this.story}</span>
          <br />
          <span className="movie-ratings-item-key">Gist - </span>
          <span className="movie-ratings-item-value">{this.gist}</span>
        </p>
      </div>
    );
  }
}

MovieRatingsItem.propTypes = {
  title: PropTypes.string,
  blind: PropTypes.number,
  characters: PropTypes.number,
  craftsmanship: PropTypes.number,
  gist: PropTypes.string,
  sound: PropTypes.number,
  story: PropTypes.number,
};

export default MovieRatingsItem;
