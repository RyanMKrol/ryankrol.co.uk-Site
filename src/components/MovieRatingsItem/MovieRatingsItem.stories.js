import React from 'react';
import MovieRatingsItem from './MovieRatingsItem';

export default {
  title: 'Components/MovieRatingsItem',
};

/**
 * default story
 *
 * @returns {string} JSX for the component I want to display
 */
export const normal = () => (
  <div>
    <MovieRatingsItem
      title={'300'}
      blind={86}
      characters={81}
      craftsmanship={83}
      gist={'300 Spartans dunk on a Persian weirdo'}
      sound={81}
      story={89}
      date={'07-04-2021'}
    />
  </div>
);
