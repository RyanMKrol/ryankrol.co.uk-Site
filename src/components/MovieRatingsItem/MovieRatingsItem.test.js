import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import MovieRatingsItem from './MovieRatingsItem';

const defaultArgs = {
  title: '300',
  blind: 86,
  characters: 81,
  craftsmanship: 83,
  gist: '300 Spartans dunk on a Persian weirdo',
  sound: 81,
  story: 89,
  date: '07-04-2021',
};

// Render Tests

it('renders MovieRatingsItem correctly', () => {
  const movieRatingsItem = shallow(<MovieRatingsItem {...defaultArgs} />);
  expect(toJson(movieRatingsItem)).toMatchSnapshot();
});

it('throws when required parameters are omitted', () => {
  expect(() => mount(<MovieRatingsItem />)).toThrow();
});

it('does not throw when required parameters are provided', () => {
  expect(() => mount(<MovieRatingsItem {...defaultArgs} />)).not.toThrow();
});
