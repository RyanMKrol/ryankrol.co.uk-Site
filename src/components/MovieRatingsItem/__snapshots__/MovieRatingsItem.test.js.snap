// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieRatingsItem correctly 1`] = `
<div
  className="MovieRatingsItem"
>
  <p
    className="album-ratings-item-content"
  >
    <br />
    <span
      className="album-ratings-item-key"
    >
      Title - 
    </span>
    <span
      className="album-ratings-item-value"
    >
      300
    </span>
    <br />
    <span
      className="album-ratings-item-key"
    >
      Blind - 
    </span>
    <span
      className="album-ratings-item-value"
    >
      86
    </span>
    <br />
    <span
      className="album-ratings-item-key"
    >
      Characters - 
    </span>
    <span
      className="album-ratings-item-value"
    >
      81
    </span>
    <br />
    <span
      className="album-ratings-item-key"
    >
      Craftsmanship - 
    </span>
    <span
      className="album-ratings-item-value"
    >
      83
    </span>
    <br />
    <span
      className="album-ratings-item-key"
    >
      Sound - 
    </span>
    <span
      className="album-ratings-item-value"
    >
      81
    </span>
    <br />
    <span
      className="album-ratings-item-key"
    >
      Story - 
    </span>
    <span
      className="album-ratings-item-value"
    >
      89
    </span>
    <br />
    <span
      className="album-ratings-item-key"
    >
      Gist - 
    </span>
    <span
      className="album-ratings-item-value"
    >
      300 Spartans dunk on a Persian weirdo
    </span>
    <br />
    <span
      className="album-ratings-item-key"
    >
      Date - 
    </span>
    <span
      className="album-ratings-item-value"
    >
      07-04-2021
    </span>
  </p>
</div>
`;
